open-hackathon-server/src/hackathon/hack/hackathon_manager.py

```diff
@@ -480,6 +488,79 @@ def list_hackathon_awards(self, hackathon):
         awards = hackathon.award_contents.order_by(Award.level.desc()).all()
         return [a.dic() for a in awards]
 
+    def get_hackathon_notice(self, notice_id):
+        hackathon_notice = self.db.get_object(HackathonNotice, notice_id)
+        return hackathon_notice.dic()
+
+    def create_hackathon_notice(self, hackathon, body):
+        return self.__create_hackathon_notice(hackathon, body.get('event', HACK_NOTICE_EVENT.MANUAL), body)
+
+    def update_hackathon_notice(self, hackathon, body):
+        hackathon_notice = self.db.get_object(HackathonNotice, body.get("id"))
+        if not hackathon_notice:
+            return not_found("hackathon_notice not found")
+        if hackathon.id != body.get("hackathon_id"):
+            return forbidden()
+        
+        hackathon_notice.content = body.get("content", hackathon_notice.content)
+        hackathon_notice.link = body.get("link", hackathon_notice.link)
+        hackathon_notice.update_time = self.util.get_now()
+
+        self.db.commit()
+        return hackathon_notice.dic()
+
+    def delete_hackathon_notice(self, hackathon, notice_id):
+        self.db.delete_all_objects_by(HackathonNotice, hackathon_id=hackathon.id, id=notice_id)
+        return ok()
+
+    def get_hackathon_notice_list(self, body):
+        query = HackathonNotice.query
+
+        hackathon_name = body.get("hackathon_name")
+        notice_type = body.get("type")
+        notice_event = body.get("event")
+        order_by = body.get("order_by", "time") 
+        show = body.get("show")
+
+        #filter by hackathon_name, type or event
+        if hackathon_name:
+            hackathon = self.get_hackathon_by_name(hackathon_name)
+            if hackathon:
+                query = query.filter(HackathonNotice.hackathon_id == hackathon.id)
+            else:
+                return []
+        if notice_type:
+            query = query.filter(HackathonNotice.type == int(notice_type))
+        if notice_event:
+            query = query.filter(HackathonNotice.event == int(notice_event))
```

Comment:
Does it mean that only certain `notice_event` or `all` events are selected? How can we select two or three different `notice_event`?
Similar question about `notice_type`.